domain/consensus/processes/blockbuilder/test_block_builder.go

```diff
@@ -136,6 +136,11 @@ func (bb *testBlockBuilder) buildBlockWithParents(parentHashes []*externalapi.Do
 		return nil, nil, err
 	}
 
+	_, err = bb.difficultyManager.UpdateDAADataAndReturnDifficultyBits(tempBlockHash)
```

Comment:
Here too, I'd expect this to get updated every time we add a block, then retrieved by buildBlockWithParents, not calculated every time again and again.

Reply:
Alternatively, since this is calculated each and every time it's accessed, maybe we just shouldn't store it?

Reply:
The DAA score is calculated incrementally, so we must store it